packages/flutter_tools/lib/src/ios/plist_utils.dart

```diff
@@ -2,39 +2,46 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import '../base/context.dart';
 import '../base/file_system.dart';
 import '../base/process.dart';
+import '../convert.dart';
+import '../globals.dart';
 
-const String kCFBundleIdentifierKey = 'CFBundleIdentifier';
-const String kCFBundleShortVersionStringKey = 'CFBundleShortVersionString';
-const String kCFBundleExecutable = 'CFBundleExecutable';
-
-// Prefer using [iosWorkflow.getPlistValueFromFile] to enable mocking.
-String getValueFromFile(String plistFilePath, String key) {
-  // TODO(chinmaygarde): For now, we only need to read from plist files on a mac
-  // host. If this changes, we will need our own Dart plist reader.
-
-  // Don't use PlistBuddy since that is not guaranteed to be installed.
-  // 'defaults' requires the path to be absolute and without the 'plist'
-  // extension.
-  const String executable = '/usr/bin/defaults';
-  if (!fs.isFileSync(executable))
-    return null;
-  if (!fs.isFileSync(plistFilePath))
-    return null;
-
-  final String normalizedPlistPath = fs.path.withoutExtension(fs.path.absolute(plistFilePath));
-
-  try {
-    final List<String> args = <String>[
-      executable, 'read', normalizedPlistPath,
-    ];
-    if (key != null && key.isNotEmpty) {
-      args.add(key);
+class PlistUtils {
+  const PlistUtils();
+
+  static const String kCFBundleIdentifierKey = 'CFBundleIdentifier';
+  static const String kCFBundleShortVersionStringKey = 'CFBundleShortVersionString';
+  static const String kCFBundleExecutable = 'CFBundleExecutable';
+
+  static PlistUtils get instance => context.get<PlistUtils>() ?? const PlistUtils();
+
+  Map<String, dynamic> parseFile(String plistFilePath) {
+    assert(plistFilePath != null);
+    const String executable = '/usr/bin/plutil';
```

Comment:
What about
```
final String executable = os.which('plutil');
if (executable == null)
  return null;
```
or throw
```

Reply:
I'd rather hard-code `/usr/bin/plutil` so we don't get unpredictable behavior if someone has another plutil in their bath.  I updated it to throw.